Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsViewModel.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.samples.crane.details
+
+import androidx.compose.samples.crane.data.DestinationsRepository
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.squareup.inject.assisted.Assisted
+import com.squareup.inject.assisted.AssistedInject
+import com.squareup.inject.assisted.dagger2.AssistedModule
+import dagger.Module
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ActivityRetainedComponent
+
+class DetailsViewModel @AssistedInject constructor(
+    destinationsRepository: DestinationsRepository,
+    @Assisted cityName: String
+) : ViewModel() {
+
+    val cityDetails = destinationsRepository.getDestination(cityName)
```

Comment:
Can you add the type here? Type inference isn't great for public fields/functions.

Is this a LiveData? If so, this the UI state stream should never be `null`. The ViewModel should take care of handling the null case, not the UI.

Reply:
Done! Thank you